### PR TITLE
MP-74 Analytics for collections v2. 

### PR DESCRIPTION
**Pull request by Guyzeroth**
## Summary

- Added click metrics for collections v2.0
- Added page request for all page in app. 
- Toggles on settings now have metrics attached.

### CODE REVIEW

File: client/src/telemetry/glean-context.tsx [hunk 1]
```diff
@@ -93,3 +96,22 @@ export function GleanProvider(props: { children: React.ReactNode }) {
...
+  }
+  return useEffect(
+    () =>
+      gleanAnalytics.page({
+        path: window?.location.toString(),
+        referrer: document?.referrer,
+      }),
+    [referrer, location, loc.pathname]
+  );
+}
```
Comment (LeoMcA) [inline client/src/telemetry/glean-context.tsx:117]: I'm a little confused by `referrer` and `location` being defined, and then not used in the glean call, and I think the way you're filtering out server-side events is causing the hook to be called multiple times if components are loaded and unloaded (as they are in local dev to prepare for react doing some fancy caching stuff in v19 or something).

How about something like this?

```js
export function useGleanPage() {
  const isServer = useIsServer();
  const routerLocation = useLocation();

  const referrer = document?.referrer;
  const location = window?.location.toString();

  return useEffect(() => {
    if (!isServer) {
      gleanAnalytics.page({
        path: location,
        referrer,
      });
    }
  }, [isServer, routerLocation.pathname, referrer, location]);
}
```

Comment (Guyzeroth) [reply client/src/telemetry/glean-context.tsx:117]: Good call out. I think initially I had some issues with document/window being undefined and this was a work around. these globals are excluded from being required in the dependency array anyways.

File: client/src/telemetry/constants.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export const ARTICLE_ACTIONS_COLLECTION_SELECT_OPENED =
```
[the remark this answers is not in the document]
Comment (Guyzeroth) [reply client/src/telemetry/constants.ts:1]: I agree lower case. As someone who may setup dashboards and need to answer quite questions of the data I actually like them in one place as an easy reference to what is currently being recorded. I don't think a soft schema is necessary unless we find them to become unwieldy.

File: client/src/banners/active-banner.tsx [hunk 3]
```diff
@@ -109,14 +115,29 @@ function PlusLaunchAnnouncementBanner({
...
   const sendCTAEventToGA = useSendCTAEventToGA();
-
+  const glean = useGlean();
+  const user = useUserData();
+  const _onDismissed = onDismissed;
+  onDismissed = () => {
+    glean.click({
+      source: BANNER_PREVIEW_FEATURES_DISMISSED,
+      subscription_type: user?.subscriptionType || "none",
+    });
```
[the remark this answers is not in the document]
Comment (Guyzeroth) [reply client/src/banners/active-banner.tsx:125]: I'll extract it to a function for better readability. Using a hook would not make sense here IMO

**Pull request merged**